### PR TITLE
Scan summary drop package verification code

### PR DESCRIPTION
**Pull request by fviernau**
The SPDX package verification codes have started to turn obsolete once `ScanSummary` started to correspond (also) to a `Provenance`, as of the re-write of the scanner to scan "by provenance". However, scan summaries are still used to describe results of a package, e.g. as `OrtResult.getScanResult()` does. Recently reporters have been updated to compute these verification codes solely based on the file lists contained in `OrtResult` turning `ScanSummary.packageVerificationCode` fully obsolete. This PR removes that property.

Follow-up for #6948.

### CODE REVIEW

File: cli/src/funTest/assets/semver4j-ort-result.yml [hunk 1]
```diff
@@ -376,7 +376,7 @@ advisor:
       - advisor:
           name: "NexusIQ"
           capabilities:
-            - "VULNERABILITIES"
+          - "VULNERABILITIES"
```
Comment (sschuberth) [inline cli/src/funTest/assets/semver4j-ort-result.yml:377]: BTW, this is interesting in the context of https://github.com/oss-review-toolkit/ort/issues/7219. Actually both indentations are valid YAML (so it's not really a formatting issue that needs fixing). I'm curious to know what change in ORT causes the different serialization... probably a Jackson / SnakeYAML update.

File: scanner/src/main/kotlin/Scanner.kt [hunk 2]
```diff
@@ -777,8 +771,7 @@ fun ScanResult.toNestedProvenanceScanResult(nestedProvenance: NestedProvenance):
 
     val provenances = nestedProvenance.getProvenances()
     val scanResultsByProvenance = provenances.associateWith { provenance ->
-        // TODO: Find a solution for the incorrect packageVerificationCode and for how to associate issues to the
-        //       correct scan result.
+        // TODO: Find a solution for how to associate issues to the correct scan result.
```
Comment (sschuberth) [inline scanner/src/main/kotlin/Scanner.kt:774]: Isn't the change of this comment actually a fixup for calculating the PVC on the fly?

File: scanner/src/funTest/kotlin/storages/AbstractStorageFunTest.kt [hunk 3]
```diff
@@ -155,7 +154,9 @@ abstract class AbstractStorageFunTest(vararg listeners: TestListener) : WordSpec
 
             "not store a result for the same scanner and provenance twice" {
                 val summary1 = scanSummaryWithFiles
-                val summary2 = scanSummaryWithFiles.copy(packageVerificationCode = "anotherPackageVerificationCode")
+                val summary2 = scanSummaryWithFiles.copy(
+                    startTime = scanSummaryWithFiles.startTime.plusSeconds(10)
```
Comment (sschuberth) [inline scanner/src/funTest/kotlin/storages/AbstractStorageFunTest.kt:158]: Unrelated change?

File: scanner/src/main/kotlin/Scanner.kt (hunk 2 above)
Comment (sschuberth) [reply scanner/src/main/kotlin/Scanner.kt:774]: BTW, while double-checking I also discovered the comment at https://github.com/oss-review-toolkit/ort/blob/70a13d2d70c7c54ddcc446b74580e459c01883ff/model/src/main/kotlin/utils/ScanResultUtils.kt#L37-L38 which could also be adjusted now.

Comment (sschuberth) [reply scanner/src/main/kotlin/Scanner.kt:774]: And also https://github.com/oss-review-toolkit/ort/blob/70a13d2d70c7c54ddcc446b74580e459c01883ff/scanner/src/main/kotlin/provenance/NestedProvenanceScanResult.kt#L67-L68

File: scanner/src/funTest/kotlin/storages/AbstractStorageFunTest.kt (hunk 3 above)
Comment (fviernau) [reply scanner/src/funTest/kotlin/storages/AbstractStorageFunTest.kt:158]: This change was intended. IIUC the removed code intended to create a second summary which is different from the first one. In particular it was only targeting difference in general - not difference in terms of package verification code. This is why I choose to achive the (assumed) goal by modifying the timestamp instead.

File: scanner/src/main/kotlin/Scanner.kt (hunk 2 above)
Comment (fviernau) [reply scanner/src/main/kotlin/Scanner.kt:774]: > Isn't the change of this comment actually a fixup for calculating the PVC on the fly?

Maybe it depends on the point of view. If one say the scanner does not have any knowledge about any "reporters", which I believe is true, then probably not. Also my approach was: 1. Implement a replacement for the non working existing package verification code first (previous PR) 2. Then remove all the left-over in the scanner (this PR). That separation still makes sense to me, so I'd say it's not a fix-up.

File: cli/src/funTest/assets/semver4j-ort-result.yml (hunk 1 above)
Comment (fviernau) [reply cli/src/funTest/assets/semver4j-ort-result.yml:377]: > Actually both indentations are valid YAML

I prefer to use what ORT outputs programatically as the single source of truth what is the right intendation. Otherwise, we could not use our own output anymore without reformatting it causing manual effort.

Comment (sschuberth) [reply cli/src/funTest/assets/semver4j-ort-result.yml:377]: > I prefer to use what ORT outputs programatically

My point is: What ORT outputs programmatically seems to have changed (at some point of time). Anyway, I prefer the "new" syntax with less indentation, so I like the change per se.

File: scanner/src/funTest/kotlin/storages/AbstractStorageFunTest.kt (hunk 3 above)
Comment (sschuberth) [reply scanner/src/funTest/kotlin/storages/AbstractStorageFunTest.kt:158]: I see. This wasn't obvious to me, but it makes sense.

**Review by sschuberth — APPROVED** [review]

File: scanner/src/funTest/kotlin/storages/AbstractStorageFunTest.kt (hunk 3 above)
Comment (fviernau) [reply scanner/src/funTest/kotlin/storages/AbstractStorageFunTest.kt:158]: yeah, it would have been better I mentioned it in the commit.
